packages/v2/contracts/ANTv2.sol

```diff
@@ -0,0 +1,159 @@
+pragma solidity 0.5.17;
+
+import './interfaces/IERC20.sol';
+import './libraries/SafeMath.sol';
+
+
+// Lightweight token modelled after UNI-LP: https://github.com/Uniswap/uniswap-v2-core/blob/v1.0.1/contracts/UniswapV2ERC20.sol
+// Adds:
+//   - A minting role
+//   - transferWithAuth()
+contract ANTv2 is IERC20 {
+    using SafeMath for uint256;
+
+    string private constant ERROR_NOT_MINTER = "ANTV2:NOT_MINTER";
+    string private constant ERROR_AUTH_EXPIRED = "ANTV2:AUTH_EXPIRED";
+    string private constant ERROR_AUTH_NOT_YET_VALID = "ANTV2:AUTH_NOT_YET_VALID";
+    string private constant ERROR_AUTH_ALREADY_USED = "ANTV2:AUTH_ALREADY_USED";
+    string private constant ERROR_INVALID_SIGNATURE = "ANTV2:INVALID_SIGNATURE";
+
+    string public constant name = "Aragon Network Token";
+    string public constant symbol = "ANT";
+    uint8 public constant decimals = 18;
+
+    address public minter;
+    uint256 public totalSupply;
+    mapping (address => uint256) public balanceOf;
+    mapping (address => mapping(address => uint256)) public allowance;
+
+    bytes32 public DOMAIN_SEPARATOR;
```

Comment:
Having this variable as in the implementation using `EIP712Domain` inheritance does make sence?

Reply:
Do you mean having a separate contract to do this, like EIP3009's sample implementation?

I didn't find a good reason to do it there; for the sake of brevity and ease-of-audit, I opted to inline rather than add inheritance.